src/platform/ESP32/ESP32FactoryDataProvider.h

```diff
@@ -60,6 +60,24 @@ class ESP32FactoryDataProvider : public CommissionableDataProvider,
     CHIP_ERROR GetProductAttestationIntermediateCert(MutableByteSpan & outBuffer) override;
     CHIP_ERROR SignWithDeviceAttestationKey(const ByteSpan & messageToSign, MutableByteSpan & outSignBuffer) override;
 
+#ifdef CONFIG_ENABLE_SET_CERT_DECLARATION_API
+    /**
+     * API to set the CD.
+     *
+     * GetCertificationDeclaration() API impl reads the CD from the NVS namespace `chip-factory`.
+     * Use this API to set the CD if it is stored at a different place, eg: embedded in the firmware.
+     * Subsequent reads after calling this API will return the set CD.
+     *
+     * API do not make a copy of underlying data, and any calls made by the Matter stack shall reach it.
```

Comment:
I don't understand what the "shall reach it" part is trying to say.

Reply:
I've adjusted the API doc, please take a look.